test/test_connectionpool.py

```diff
@@ -225,6 +228,32 @@ def test_max_connections(self) -> None:
 
             assert pool.num_connections == 1
 
+    def test_put_conn_message_when_pool_is_full_nonblocking(self, caplog: Any) -> None:
+        """
+        If maxsize = n and we _put_conn n + 1 conns, the n + 1th conn will
+        get closed and will not get added to the pool.
+        """
+        caplog.set_level(logging.WARNING)
+        with HTTPConnectionPool(host="localhost", maxsize=1, block=False) as pool:
+            conn1 = pool._get_conn()
+            # pool.pool is empty because we popped the one None that pool.pool was initialized with
+            # but this pool._get_conn call will not raise EmptyPoolError because block is False
+            conn2 = pool._get_conn()
+
+            with patch.object(conn1, "close") as conn1_close:
+                with patch.object(conn2, "close") as conn2_close:
+                    pool._put_conn(conn1)
+                    pool._put_conn(conn2)
+
+            assert conn1_close.called is False
+            assert conn2_close.called is True
+
+            assert conn1 == pool._get_conn()
+            assert conn2 != pool._get_conn()
+
+            # Primary objective is making sure the pool size is explicitly listed
+            assert "Connection pool size: 1" in caplog.text
```

Comment:
Can you please just modify the next test to add this line just after `assert conn2_close.called is True`? And remove this new test.

Reply:
Sure thing! I originally made it separate because I thought you wanted everything to be completely decoupled (in terms of test purposes), but I agree it makes sense here